tests/core/test_configuration.py

```diff
@@ -80,6 +80,15 @@ def test_case_sensitivity(self):
         assert conf.get("core", "PERCENT") == "with%inside"
         assert conf.get("CORE", "PERCENT") == "with%inside"
 
+    def test_config_as_dict(self):
+        """Test that getting config as dict works even if
+        environment has non-legal env vars"""
+        with unittest.mock.patch.dict('os.environ'):
+            os.environ['AIRFLOW__VAR__broken'] = "not_ok"
+            asdict = conf.as_dict(raw=True, display_sensitive=True)
+        assert asdict.get('VAR') is None
```

Comment:
This is not the behaviour I would expect. It should instead still be present.

There are a few plugins out there that add their own config sections, and that is something we want to support.

Reply:
> This is not the behaviour I would expect. It should instead still be present.
> 
> There are a few plugins out there that add their own config sections, and that is something we want to support.

If they do and do not enforce ALL_UPPER_CASE they will break if using `as_dict()` with the current code, with this patch they will be ignored but not crash. I don't know that allowing mixed case env vars is wanted?
The only functional change here is to not crash if `is_dict` is called while there is mixed case env vars (starting with `AIRFLOW__` but then not all uppercase).

Reply:
@ashb @XD-DENG I believe the PR behaviour is conservative (fixing the crash, not adding behaviour). Adding the inclusion of the MIXED_case env vars without breaking other things in the configuration module looks like it would take larger (in scope) changes.

Reply:
I'll take a look on Monday if XD doesn't get to it first

Reply:
Ping @ashb @XD-DENG